parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -379,42 +378,40 @@ public DeleteWriteBuilder equalityFieldIds(int... fieldIds) {
           appenderBuilder.build(), FileFormat.PARQUET, location, spec, partition, keyMetadata, equalityFieldIds);
     }
 
-
-    public <T> PositionDeleteWriter<T> buildPositionWriter() throws IOException {
+    public <T> PositionDeleteWriter<T> buildPositionWriter(ParquetValueWriters.PathPosAccessor<?, ?> accessor)
```

Comment:
I'd rather not add this to the build method. Nothing distinguishes it from other options, so I think we should add transformations as configuration methods, like we do for `equalityFieldIds`.

I'm also thinking that it would be good to have a more light-weight way to add these transforms. Rather than an additional accessor that has two abstract methods, why not just register functions? It could look like this:

```java
  Avro.writeDeletes(outFile)
      ...
      .transformPaths(StringData::fromString)
      .buildPositionWriter();
```

That way it's easier to use a method reference rather than creating a class. And nothing actually needs to transform pos yet, so we can just leave that out.

Reply:
I like the idea about registering a light-weight function to convert the `CharSequence` to `StringData`.